### DISCUSSION TITLE
Can I build a AIR, IPA or APK file as a Free Tier ?

### DISCUSSION THREAD

**Discussion by oscarduvar** [airsdk/Adobe-Runtime-Support#918] [Q&A]
Hello,

I have downloaded the AIR SDK for Windows - (33.1.1.476)  via https://airsdk.harman.com/download
I have overlaid the Flex SDK 16.1 - AIR31 on my computer (Windows 10).
I managed to launch my Flex projects with my FlashBuilder 4.6 (and see the Harman splashscreen as excepted).
But once I try to build either an AIR file, an IPA or an APK file I am stucked on the last page of the build process (certificate/ANE page), without any error.

That's because I have no license ?

Thanks for your help.

**Comment by leossmith** [comment]
That is FlashBuilder being older than life itself and breaking. I never managed to make it work with the new SDK. Jumped over to Intellij solving all my problems.

**Reply by ajwfrost** [reply]
It's most likely because Flash Builder includes a JRE using version 1.6, but we needed to update the AIR Developer Tool to use 1.7 because of the new Android requirements... So if you update the JRE it may start working (look for a 'jre' subfolder of the Flash Builder installation location, and replace this with the jre from another Java installation e.g. 1.7 or 1.8...)

But yes, you may be better off migrating to an IDE like Intellij or Moonshine..

**Reply by oscarduvar** [reply]
Thanks,

I have forced my 32-bits Flash Builder 4.6 to use a 1.8 JRE (also 32-bits) (with -vm in the FlashBuilder.ini)
Then I suceeded to build an IPA and pushed it on my iPad with iOS 15 but I still have the message from Apple asking to use the latest code signature Format

Let's see with another IDE

**Reply by ajwfrost** [reply]
> the message from Apple asking to use the latest code signature Format

Can you confirm what you're currently using as the format? (you can unzip the .ipa file and then run `codesign -dvvv Payload/application.app` to find out, e.g. 20400)

And what's the exact message from Apple?

thanks
